### PR TITLE
chore: add readonly accessors to all resources

### PR DESCRIPTION
**Pull request by eladb**
Make sure that all resources have read-only accessors for all properties
and addXxx methods for collections.

---

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/cdk8s-plus/CONTRIBUTING.md [hunk 1]
```diff
@@ -12,29 +12,112 @@ Even so, there are several design principles we would like to try and maintain:
...
+     public get baz() { return { ...this._baz }; }
+   }
+   ```
 
-1. Kuberenetes spec objects have an extremely large surface area. Configuring all properties in the constructor can easily result in cluttered
-and un-readable code. Exposing post instantiation methods can provide a cleaner, more semantic mechanism of configuration.
 
-2. This library aims to reduce the complexity of creating kuberenetes configuration files. An important enabler to achieve this,
-is the ability to auto wire different resources to work together. To that end, post instantiation mutations are often required.
+The reasoning behind this is two fold:
```
Comment (iliapolo) [inline packages/cdk8s-plus/CONTRIBUTING.md:103]:
```suggestion
The reasoning behind this is three fold:
```

Or better yet:

```suggestion
There are several reasonings behind this:
```

File: packages/cdk8s-plus/src/config-map.ts [hunk 2]
```diff
@@ -91,7 +91,16 @@ export class ConfigMap extends Resource implements IConfigMap {
...
     this.verifyKeyAvailable(key);
 
-    this.data[key] = value;
+    this._data[key] = value;
+  }
+
+  /**
+   * The data associated with this config map.
+   *
+   * Returns an immutable copy.
```
Comment (iliapolo) [inline packages/cdk8s-plus/src/config-map.ts:100]: Not exactly accurate, its just a copy, but still mutable. We should probably create `ImmutableList` and `ImmutableMap` classes later on.

Comment (eladb) [reply packages/cdk8s-plus/src/config-map.ts:100]: Okay, I will think of better phrasing. Maybe `Returns a copy. To add entries use "addData()"`?

Comment (eladb) [reply packages/cdk8s-plus/src/config-map.ts:100]: Fixed

**Pull request merged**